R/GetPostDeleteDefinition.R

```diff
@@ -302,18 +302,21 @@ updateDefinition <- function(definition, baseUrl, category) {
     stop(paste(definition$name, "Is not a valid cohort name."))
   }
 
-  # Check definition is ok
-  if (category == "cohort") {
-    jsonExpression <- .toJSON(definition$expression)
-    checkUrl <- paste(baseUrl, argument$categoryUrl, "check", sep = "/")
+  # # Check definition is ok 
   
-    tryCatch({
-      response <- .postJson(checkUrl, jsonExpression)
-      content <- httr::content(response)
-    }, error = function (error) {
-        stop(paste("Error with", category, "definition:", error))
-    })
-  }
+  ## why??  can't you save a definition with errors?
```

Comment:
I'm happy to allow cohorts to be posted if there are errors (as long as ATLAS allows this - there should be consistency between the javascript and R clients), but the user should be presented with the warnings/notes when this happens.

Reply:
I'm confused. This tryCatch is handling http errors which seem different than errors with the cohort. For example, error code of 500 "internal server error" should not mean "the cohort definition has a problem but the POST operation succeeded" right?

Reply:
@chrisknoll  :) cant understand your concern. 

This is not checking if json has errors. It is only checking if it posted it.  All this try catch is doing is - replacing default error message with something different

Reply:
> There is centralized http error code handling in the .request function which is being overridden here.

This function was written before the support for security. When we decided to support security in ROhdsiWebApi - we did not remove these old error handling